### DISCUSSION TITLE
Php image magic Eps to png getting blurry image not getting actual image .

### DISCUSSION THREAD

**Discussion by imprintnext-sanjeeb** [ImageMagick/ImageMagick#6164] [Development]
### ImageMagick version

Version: ImageMagick 6.9.10-23 Q16 x86_64 20190101

### Operating system, version and so on

Ubuntu

### Description

While to convert eps file to png , 
Not getting actual image also getting blurry image .
[2ndSkin-Logo-2couleurs.zip](https://github.com/ImageMagick/ImageMagick/files/10979153/2ndSkin-Logo-2couleurs.zip)

**Command using to convert** 

$path = '2ndSkin-Logo-2couleurs.eps';
$save_path = "magic/result.png";
shell_exec('convert -colorspace sRGB ' . $path .  " -resize 100% PNG32:" . $save_path );

**Result that i have get** 

![result](https://user-images.githubusercontent.com/88526225/225299407-0482a3b2-41fa-4732-a8ca-8b453883c05a.png)

**But the actual result will be** 

![210345369-38c1ff03-2e3b-4fea-bcf4-011de7507070](https://user-images.githubusercontent.com/88526225/225299651-2d15fe39-c3d7-40e0-b5af-2db56d62dfaa.png)

**Comment by fmw42** [comment]
Your IM version is very old. I suggest you upgrade.

With IM 6.9.12-81, I get

convert 2ndSkin-Logo-2couleurs.eps -background white -flatten -trim +repage -bordercolor white -border 0 x.png

![x](https://user-images.githubusercontent.com/13458218/225357776-1b998df8-71bd-4a9c-8300-21c8aeb5106c.png)

<br>

**What is the point of -border 0?**

**Comment by snibgo** [comment]
With IM v7.1.0-49 and Ghostscript v10.0, I get the same result as fmw42.

@imprintnext-sanjeeb: what version of Ghostscript do you have? If it is old, I suggest an upgrade.

**Comment by imprintnext-sanjeeb** [comment]
I have update the ghost script and imagic 
Ghost - 10.00.0
Magic - Version: ImageMagick 7.1.1-4
still getting blurry image 
command i have used 
shell_exec('convert -colorspace sRGB -flatten -quality 100 -size "1024x1024" ' . $path .  " " . $save_path );

![result (1)](https://user-images.githubusercontent.com/88526225/225635233-754a3ded-8479-4872-931b-388e6459f5ea.png)

**Reply by fmw42** [reply]
Your syntax is wrong for IM 7. You should read the input before any processing. Also for IM 7 use magick not convert. So try

`magick input.suffix -colorspace sRGB -flatten -resize "1024x1024" -quality 100 output.suffix`


Note to resize, use -resize not -size

**Comment by imprintnext-sanjeeb** [comment]
Command i have used 

magick /var/www/html_php81-8310/magic/2ndSkin-Logo-2couleurs.eps -colorspace sRGB -flatten -resize '1024x1024' -quality 100 /var/www/html_php81-8310/magic/result.png

But i have got same result
![result (2)](https://user-images.githubusercontent.com/88526225/225822294-aa11b030-32a9-4c5d-8d66-07c7e62b7090.png)

**Comment by fmw42** [comment]
On IM 7.1.1-3 I get your same result.

`magick 2ndSkin-Logo-2couleurs.eps -colorspace sRGB -flatten -resize '1024x1024' -quality 100 result.png`

**magick: Invalid TIFF directory; tags are not sorted in ascending order. `TIFFReadDirectoryCheckOrder' @ warning/tiff.c/TIFFWarnings/951.**

Your embedded TIFF file has a warning. I am not sure if that is the cause of the effect, but I get your same result. It may be an issue with libtiff.

![result](https://user-images.githubusercontent.com/13458218/225953680-9cf6a21e-71cd-4fbf-9a24-5e9f43f95e17.png)

But magick identify -verbose says your file has spot colors and Imagemagick does not know how to handle those. I am not sure if this is an issue or not.
